packages/sonarwhal/src/lib/config.ts

```diff
@@ -190,16 +190,18 @@ const loadIgnoredUrls = (userConfig: UserConfig): Map<string, RegExp[]> => {
 /** Validates that the given configuration for a rule is valid */
 const validateRules = (rulesConfig: RulesConfigObject, userConfig: UserConfig) => {
     const rules = Object.keys(rulesConfig);
-
-    rules.forEach((rule) => {
+    const invalidRuleConfigs = rules.reduce((invalidRules, rule) => {
         const Rule = resourceLoader.loadRule(rule, userConfig.extends);
-
         const valid: boolean = validateRule(Rule.meta, userConfig.rules[rule], rule);
 
         if (!valid) {
-            throw new Error(`Rule ${rule} has an invalid configuration`);
+            invalidRules.push(rule);
         }
-    });
+
+        return invalidRules;
+    }, []);
+
+    return invalidRuleConfigs;
```

Comment:
I think I would change the return of this to be more aligned with what we do in other places and also so the name of the function is still valid. Maybe return something like:
```js
{
  valid: [],
  invalid: []
}
```